I expect
```
C:\projects\source\build\core\unit_test\serial\TestSerial_MathematicalFunctions.cpp : fatal error C1128: number of sections exceeded object file format limit: compile with /bigobj [C:\projects\source\build\core\unit_test\KokkosCore_UnitTest_Serial1.vcxproj]
``` 
in the Windows CI (and similarly in the HPX CI) to be resolved by #4818.

--- a/core/src/Kokkos_Serial.hpp
+++ b/core/src/Kokkos_Serial.hpp
@@ -570,7 +570,7 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
             : pointer_type(
                   internal_instance->m_thread_team_data.pool_reduce_local());
 
-    typename Analysis::template Reducer<> final_reducer(
+    typename Analysis::Reducer final_reducer(
         &ReducerConditional::select(m_functor, m_reducer));
 
     reference_type update = final_reducer.init(ptr);
@@ -665,7 +665,7 @@ class ParallelScan<FunctorType, Kokkos::RangePolicy<Traits...>,
         pool_reduce_size, team_reduce_size, team_shared_size,
         thread_local_size);
 
-    typename Analysis::template Reducer<> final_reducer(&m_functor);
+    typename Analysis::Reducer final_reducer(&m_functor);
 
     reference_type update = final_reducer.init(pointer_type(
         internal_instance->m_thread_team_data.pool_reduce_local()));
@@ -729,7 +729,7 @@ class ParallelScanWithTotal<FunctorType, Kokkos::RangePolicy<Traits...>,
         pool_reduce_size, team_reduce_size, team_shared_size,
         thread_local_size);
 
-    typename Analysis::template Reducer<> final_reducer(&m_functor);
+    typename Analysis::Reducer final_reducer(&m_functor);
 
     reference_type update = final_reducer.init(pointer_type(
         internal_instance->m_thread_team_data.pool_reduce_local()));
@@ -868,7 +868,7 @@ class ParallelReduce<FunctorType, Kokkos::MDRangePolicy<Traits...>, ReducerType,
             : pointer_type(
                   internal_instance->m_thread_team_data.pool_reduce_local());
 
-    typename Analysis::template Reducer<> final_reducer(
+    typename Analysis::Reducer final_reducer(
         &ReducerConditional::select(m_functor, m_reducer));
 
     reference_type update = final_reducer.init(ptr);
@@ -1056,7 +1056,7 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
             : pointer_type(
                   internal_instance->m_thread_team_data.pool_reduce_local());
 
-    typename Analysis::template Reducer<> final_reducer(
+    typename Analysis::Reducer final_reducer(
         &ReducerConditional::select(m_functor, m_reducer));
 
     reference_type update = final_reducer.init(ptr);

--- a/core/src/impl/Kokkos_FunctorAnalysis.hpp
+++ b/core/src/impl/Kokkos_FunctorAnalysis.hpp
@@ -798,10 +798,6 @@ struct FunctorAnalysis {
       return *dst;
     }
 
-    KOKKOS_INLINE_FUNCTION constexpr reference_type reference() const noexcept {
-      return Reducer::template ref<candidate_is_array>();
-    }
-
     KOKKOS_INLINE_FUNCTION constexpr int length() const noexcept {
       return Reducer::template len<candidate_is_array>();
     }

--- a/core/src/impl/Kokkos_FunctorAnalysis.hpp
+++ b/core/src/impl/Kokkos_FunctorAnalysis.hpp
@@ -760,7 +760,6 @@ struct FunctorAnalysis {
                     !Kokkos::is_reducer<Functor>::value,
                 "Reducer must have a join member function!");
 
-  template <class MemorySpace = typename execution_space::memory_space>
   struct Reducer {
    private:
     Functor const* const m_functor;
@@ -782,7 +781,6 @@ struct FunctorAnalysis {
     using reducer        = Reducer;
     using value_type     = std::remove_const_t<FunctorAnalysis::value_type>;
     using pointer_type   = value_type*;
-    using memory_space   = MemorySpace;
     using reference_type = FunctorAnalysis::reference_type;
     using functor_type   = Functor;  // Adapts a functor
 

--- a/core/unit_test/TestFunctorAnalysis.hpp
+++ b/core/unit_test/TestFunctorAnalysis.hpp
@@ -75,7 +75,7 @@ void test_functor_analysis() {
                                     Kokkos::RangePolicy<ExecSpace>,
                                     decltype(c01)>;
 
-  using R01 = typename A01::template Reducer<typename ExecSpace::memory_space>;
+  using R01 = typename A01::Reducer;
 
   static_assert(std::is_same<typename A01::value_type, void>::value, "");
   static_assert(std::is_same<typename A01::pointer_type, void>::value, "");
@@ -94,7 +94,7 @@ void test_functor_analysis() {
   using A02 = Kokkos::Impl::FunctorAnalysis<
       Kokkos::Impl::FunctorPatternInterface::REDUCE,
       Kokkos::RangePolicy<ExecSpace>, decltype(c02)>;
-  using R02 = typename A02::template Reducer<typename ExecSpace::memory_space>;
+  using R02 = typename A02::Reducer;
 
   static_assert(std::is_same<typename A02::value_type, double>::value, "");
   static_assert(std::is_same<typename A02::pointer_type, double*>::value, "");
@@ -114,7 +114,7 @@ void test_functor_analysis() {
   using A03 = Kokkos::Impl::FunctorAnalysis<
       Kokkos::Impl::FunctorPatternInterface::REDUCE,
       Kokkos::RangePolicy<ExecSpace>, TestFunctorAnalysis_03>;
-  using R03 = typename A03::template Reducer<typename ExecSpace::memory_space>;
+  using R03 = typename A03::Reducer;
 
   static_assert(std::is_same<typename A03::value_type,
                              TestFunctorAnalysis_03::value_type>::value,

